ColorHighlighter.py

```diff
@@ -1229,33 +1233,48 @@ def __init__(self, finder):
 
     def get_vars(self, view):
         color_vars_files = []
-        if is_st3():
+        if not is_st3():
+            project_settings = build_system_hack.get_project_settings(sublime.active_window())
+            wnd = sublime.active_window()
+        else:
             wnd = view.window()
-            if wnd is not None:
-                pdata = wnd.project_data()
-                if pdata is not None:
-                    color_vars_files = pdata.get("color_variables_files", [])
-                    if type(color_vars_files) is not list:
-                        color_vars_files = [color_vars_files]
-                    color_vars_file = pdata.get("color_variables_file", None)
-                    if color_vars_file is not None:
-                        color_vars_files.append(color_vars_file)
 
-                    for f in color_vars_files:
-                        self.parse_vars_file(f)
+        if wnd is not None:
+            if is_st3():
+                pdata = wnd.project_data()
+            else:
+                pdata = view.settings()
+
+            if pdata is not None:
+                color_vars_files = pdata.get("color_variables_files", [])
+                if type(color_vars_files) is not list:
+                    color_vars_files = [color_vars_files]
+                color_vars_file = pdata.get("color_variables_file", None)
+                if color_vars_file is not None:
+                    color_vars_files.append(color_vars_file)
+
+                for f in color_vars_files:
+                    if is_st3():
+                        self.parse_vars_file(sublime.expand_variables(f, wnd.extract_variables()))
```

Comment:
Calling sublime.expand_variables(f, wnd.extract_variables()) or Template(f).substitute(project_settings) should be extracted in a helper fuction expand_vadiables(f, wnd).